statsmodels/robust/covariance.py

```diff
@@ -0,0 +1,1125 @@
+        approximately consistent with the normal distribution. Rescaling is
+        based on the median of the distances and of the chisquare distribution.
+        Other options are not yet available.
+        If rescale is the string "none", then no rescaling is performed.
+
+    Returns
+    -------
+    Holder instance with attributes: cov, mean, w, dist, it, converged
+
+    Notes
+    -----
+    This iterates over calculating the mahalanobis distance and weighted
+    covariance. See Feingold and Drton 2014 for the motivation using weights
+    based on the multivariate t distribution. Note that this does not calculate
+    their alternative t distribution which requires numerical or Monte Carlo
+    integration.
+
+    References
+    ----------
+    .. [1] Finegold, Michael, and Mathias Drton. 2011. Robust graphical
+       modeling of gene networks using classical and alternative
+       t-distributions. Annals of Applied Statistics 5 (2A): 1057-80.
+
+    """
+    data = np.asarray(data)
+    nobs, k_vars = data.shape  # noqa: F841
+
+    if cov_init is None:
+        cov_init = np.cov(data.T)
+
+    converged = False
+    cov = cov_old = cov_init
+    for it in range(maxiter):
+        dist = mahalanobis(data, cov=cov)
+        w = weights_func(dist, *weights_args)
+        cov, mean = cov_weighted(data, w)
+        if np.allclose(cov, cov_old, atol=atol, rtol=rtol):
+            converged = True
+            break
+
+    if rescale == 'none':
+        s = 1
+    elif rescale == 'med':
+        s = np.median(dist) / stats.chi2.ppf(0.5, k_vars)
+        cov *= s
+    else:
+        raise NotImplementedError('only rescale="med" is currently available')
+
+    res = Holder(cov=cov, mean=mean, weights=w, mahalanobis=dist,
+                 scale_factor=s, n_iter=it, converged=converged)
+    return res
+
+
+def _cov_starting(data, is_standardized=True, quantile=0.5):
+    """compute some robust starting covariances
+
+    The returned covariance matrices are intended as starting values
+    for further processing. The main purpose is for algorithms with high
+    breakdown point.
+    The quality as standalone covariance matrices varies and might not
+    be very good.
+
+
+    """
+    x = np.asarray(data)
+    nobs, k_vars = x.shape
+    if not is_standardized:
+        # there should be a helper function/class
+        center = np.median(data, axis=0)
+        xs = (x - center)
+        std = mad0(data)
+        xs /= std
+    else:
+        center = np.median(data, axis=0)
+        xs = x - center
+
+    cov_all = []
+    d = mahalanobis(xs, cov=None, cov_inv=np.eye(k_vars))
+    percentiles = [(k_vars+2) / nobs * 100 * 2, 25, 50]
+    cutoffs = np.percentile(d, percentiles)
+    for p, cutoff in zip(percentiles, cutoffs):
+        xsp = xs[d < cutoff]
+        c = np.cov(xsp.T)
+        corr_factor = coef_normalize_cov_truncated(p / 100, k_vars)
+        c0 = Holder(cov=c * corr_factor, mean=xsp.mean(0) + center,
```

Comment:
likely a bug
should be `mean=xsp.mean(0) * std + center`  if data is standardized/scaled from original